src/Microsoft.AspNetCore.Mvc.Core/Internal/FilterActionInvoker.cs

```diff
@@ -80,14 +80,17 @@ public abstract class FilterActionInvoker : IActionInvoker
                 throw new ArgumentNullException(nameof(diagnosticSource));
             }
 
-            Context = new ControllerContext(actionContext);
 
             _controllerActionInvokerCache = controllerActionInvokerCache;
             _inputFormatters = inputFormatters;
             _valueProviderFactories = valueProviderFactories;
             Logger = logger;
             _diagnosticSource = diagnosticSource;
             _maxModelValidationErrors = maxModelValidationErrors;
+
+            Context = new ControllerContext(actionContext);
+            Context.ModelState.MaxAllowedErrors = _maxModelValidationErrors;
+            Context.ValueProviderFactories = new List<IValueProviderFactory>(_valueProviderFactories);
```

Comment:
If `ValueProviderFactories` were also a `IReadOnlyList<IValueProviderFactory>`, would this defensive copy be necessary for any scenario?